README.md

```diff
@@ -1,2 +1,209 @@
+```python
+import mir_eval
+import mir_dataset_loaders as mdl
+import numpy as np
+import sox
+
+def very_bad_melody_extractor(audio_path):
+    duration = sox.file_info.duration(audio_path)
+    time_stamps = np.linspace(0, duration, 0.01)
+    melody_f0 = np.random.uniform(low=80.0, high=800.0, size=time_stamps.shape)
+    return time_stamps, melody_f0
+
+# Evaluate on the full dataset
+orchset_scores = {}
+orchset_data = mdl.orchset.load()
+for track_id, track_data in orchset_data.items():
+    est_times, est_freqs = very_bad_melody_extractor(track_data.audio_path_mono)
+
+    ref_melody_data = track_data.melody
+    ref_times = ref_melody_data.times
+    ref_freqs = ref_melody_data.frequencies
+
+    score = mir_eval.melody.evaluate(ref_times, ref_freqs, est_times, est_freqs)
+    orchset_scores[track_id] = score
+
+# Split the results by composer and by instrumentation
+composer_scores = {}
+strings_no_strings_scores = {True: {}, False: {}}
+for track_id, track_data in orchset_data.items():
+    if track_data.composer not in composer_scores.keys():
+        composer_scores[track_data.composer] = {}
+
+    composer_scores[track_data.composer][track_id] = orchset_scores[track_id]
+    strings_no_strings_scores[track_data.contains_strings][track_id] = \
+        orchset_scores[track_id]
+
+```
+
+
+## Contributing a new dataset loader
+
+To add a new dataset loader:
+1. Create an index in `mir_dataset_loaders/indexes`, e.g. `unicorn_index.json`
+2. Create a module in `mir_dataset_loaders`, .e.g `unicorn.py`
+
+
+### Creating an index
+
+The index should be a json file where the top level keys are the unique track
+ids of the datset. The values should be a dictionary of metadata, and the keys
+of these dictionaries should be identical across the dataset.
+
+Any file path included should be relative to the top level directory of the dataset.
+
+Any unavailable field should be indicated with `null`.
+
+For example:
+```json
+{
+    'track1': {
+        'audio_path': 'Unicorn/track1.wav',
+        'color': 'red',
+        'genre': 'Electronica',
+        'sign': 'positive'
+    },
+    'track2': {
+        'audio_path': 'Unicorn/track2.wav',
+        'color': 'green',
+        'genre': 'Free Jazz',
+        'sign': 'negative'
+    },
+    'track3': {
+        'audio_path': 'Unicorn/track3.wav',
+        'color': 'purple',
+        'genre': 'Rockabilly',
+        'sign': null
+    }
+}
+```
+
+Notebooks used to create some of the dataset indexes are in the [notebooks]() folder.
+
+
+### Creating a module.
+
+copy paste this template and fill it in for your dataset:
+
+```python
+
+from collections import namedtuple
+import json
+
+from . import UNICORN_INDEX_PATH
+from .load_utils import validator
+
+UNICORN_INDEX = json.load(open(UNICORN_INDEX_PATH, 'r'))
+
+UnicornTrack = namedtuple(
+    'UnicornTrack',
+    ['track_id',
+     'TODO']
+)
+
+
+def download(data_home=None, clobber=False):
+    # TODO, write this, see orchset.py for an example
```

Comment:
Would put a nice doc string here explaining what to implement and then list `orchset.py` as example. At the very least having these parameters described here would make it a little easier